### ISSUE TITLE
Support for new Xcode 10 build system when using app extensions

### ISSUE THREAD

**Issue by blotzy** [bugsnag/bugsnag-unity#97]
### Description
The new Xcode 10 build system fails builds because of circular dependencies when a project contains extensions

### Issue
We have a rich push notification extension added via `UnityEditor.iOS.Xcode.Extensions.PBXProjectExtensions.AddAppExtension`. Removing the extension, removing the BugsnagPostProcess logic that adds the "bugsnag dsym upload script", or reverting to the legacy build system resolves the issue
```
error: Cycle in dependencies between targets 'Unity-iPhone' and 'NotificationService'; building could produce unreliable results.
Cycle path: Unity-iPhone → NotificationService → Unity-iPhone
{redacted}
○ Target 'NotificationService': Strip /Users/me/Library/Developer/Xcode/DerivedData/Unity-iPhone-bczknawwvgfuzselychrkmftqfug/Build/Intermediates.noindex/ArchiveIntermediates/Unity-iPhone/IntermediateBuildFilesPath/UninstalledProducts/iphoneos/NotificationService.appex/NotificationService
○ That command depends on command in Target 'Unity-iPhone': script phase “Run Script”
```

### Environment

* bugsnag-unity version: 3.6.5
* Unity version: 2018.2.8f1
* Operating system name and version: macOS 10.13.6, Xcode 10.0 (10A255)
* Target platform names and versions: iOS 12.0

**Comment by Cawllec** [comment]
Hi @blotzy, we're currently in the process of releasing a new version of this notifier.  I'll make sure to raise this issue during testing, and we'll add a fix if necessary.

**Comment by martin308** [comment]
@blotzy good news, I've reproduced this issue and come with a fix for it in #99 which will be included in v4 of the notifier

**Comment by avram** [comment]
So this can be closed?
